lib/SILGen/SILGenConstructor.cpp

```diff
@@ -139,9 +139,13 @@ static SubstitutionMap getSubstitutionsForPropertyInitializer(
     // replacement types are the archetypes of the initializer itself.
     return SubstitutionMap::get(
       nominal->getGenericSignatureOfContext(),
-      [&](SubstitutableType *type) {
+      [&](SubstitutableType *type) -> Type {
         if (auto gp = type->getAs<GenericTypeParamType>()) {
-          return genericEnv->mapTypeIntoContext(gp);
+          auto archetype = genericEnv->mapTypeIntoContext(gp);
+          if (!gp->isParameterPack())
+            return archetype;
+
+          return PackType::getSingletonPackExpansion(archetype);
```

Comment:
@slavapestov I tried changing this to:

```cpp
    return SubstitutionMap::get(
      nominal->getGenericSignatureOfContext(),
      genericEnv->getForwardingSubstitutionMap());
```

but it still crashed when building the standard library. I didn't spend much time figuring out why because I wanted to finish the `PackType` representation change, but I can come back to it.